### PR TITLE
Drop Python 3.8 from CI test matrix.

### PR DESCRIPTION
**Pull request by marcoesters**
### Description

`hypothesis`, which is installed to run tests, has dropped Python 3.8 support, which leads to test failures. Drop Python 3.8 from the test matrix and bump the MacOS minimum Python version to 3.9.

Additionally, run `apt-get update` before running `apt-get install` so that `apt` finds all required packages on Ubuntu.

### Checklist - did you ...

- [X] Add a file to the `news` directory ([using the template](https://github.com/conda/menuinst/blob/main/news/TEMPLATE)) for the next release's release notes?
- [X] Add / update necessary tests?
- [X] Add / update outdated documentation?

### CODE REVIEW

**Comment by jaimergp** [comment]
Do we need to update the Python requirements in pyproject.toml and the conda recipe? This is a dependency of conda, and conda-libmamba-solver has already dropped 3.8 in the upcoming v2 compatibility PR, so it's a matter of time anyway...

**Comment by jaimergp** [comment]
xref https://github.com/conda/conda/issues/14201, for context.

**Comment by marcoesters** [comment]
> Do we need to update the Python requirements in pyproject.toml and the conda recipe? 

Not immediately because the code is still Python 3.8 compatible. I would follow the guidance in the cross-referenced issue. Since it is a dependency of `conda`, we shouldn't be dropping support before they do.

**Comment by jaimergp** [comment]
Note we won't be able to provide 3.8 packages in conda-forge for the next release because it was dropped from the CI matrix.

**Comment by marcoesters** [comment]
> Note we won't be able to provide 3.8 packages in conda-forge for the next release because it was dropped from the CI matrix.

I think those are separate, albeit related, concerns and beyond the scope of this PR. It looks like the ecosystem as a whole need to come up with a deprecation policy.

**Review by larsoner — APPROVED** [review]
LGTM +1 for merge

**Comment by larsoner** [comment]
... in the meantime this fixes broken CIs which is helpful so I'll merge, even if we might want to do more in follow-ups regarding properly dropping in `pyproject.toml` and elsewhere. Thanks @marcoesters !

**Comment by jaimergp** [comment]
Thank you @larsoner for pushing this. Just for context (in this case it was ok to merge), we have been usually leaving the final merge decision to the PR author when they are a contributor already. So for the future I think it's fair to "approve and wait", and if there's something blocked by the PR itself, kindly point it out for awareness.

**Comment by marcoesters** [comment]
And since @jaimergp had questions/objections, I would have waited for his approval here. I generally want to make sure that they are all addressed.

However since he said it was okay to merge, all is well that ends well.

**Comment by larsoner** [comment]
Okay makes sense 👍

**Pull request merged**
